includes/gateways/stripe/js/connect-settings.js

```diff
@@ -0,0 +1,132 @@
+( function() {
+	function setupStripeConnectListener() {
+		onclickPreventDefault( '#wpbdp_disconnect_stripe', handleStripeDisconnectClick );
+		onclickPreventDefault( '#wpbdp_reauth_stripe', handleStripeReauthClick );
+		onclickPreventDefault( '#wpbdp_connect_with_oauth', handleConnectWithOauth );
+		renderStripeConnectSettingsButton();
+	}
+
+	function renderStripeConnectSettingsButton() {
+		var container = document.getElementById( 'wpbdp_strp_settings_container' );
+		if ( null !== container ) {
+			postAjax(
+				{
+					action: 'wpbdp_strp_connect_get_settings_button'
+				},
+				function( data ) {
+					container.innerHTML = data.html;
+				}
+			);
+		}
+	}
+
+	function onclickPreventDefault( selector, callback ) {
+		jQuery( document ).on( 'click', selector, function( event ) {
+			event.preventDefault();
+			callback( this );
+		});
+	}
+
+	function handleStripeDisconnectClick( trigger ) {
+		const testMode = isTriggerInTestMode( trigger );
+
+		const spinner = document.createElement( 'span' );
+		spinner.className = 'wpbdp-wait wpbdp_visible_spinner';
+		spinner.style.margin = 0; // The default 20px margin causes the spinner to look bad.
+		trigger.replaceWith( spinner );
+
+		strpSettingsAjaxRequest(
+			'wpbdp_stripe_connect_disconnect',
+			function() {
+				renderStripeConnectSettingsButton();
+			},
+			testMode
+		);
+	}
+
+	function handleStripeReauthClick( trigger ) {
+		strpSettingsAjaxRequest(
+			'wpbdp_stripe_connect_reauth',
+			function( data ) {
+				if ( 'undefined' !== typeof data.connect_url ) {
+					window.location = data.connect_url;
+				} else {
+					renderStripeConnectSettingsButton();
+				}
+			},
+			isTriggerInTestMode( trigger )
+		);
+	}
+
+	function handleConnectWithOauth( trigger ) {
+		trigger.classList.add( 'wpbdp-loading-button' );
+		strpSettingsAjaxRequest(
+			'wpbdp_stripe_connect_oauth',
+			function( data ) {
+				if ( 'undefined' !== typeof data.redirect_url && data.redirect_url ) {
+					window.location = data.redirect_url;
+				} else {
+					renderStripeConnectSettingsButton();
+				}
+			},
+			isTriggerInTestMode( trigger ),
+			function( data ) {
+				const messageEle = document.getElementById( 'wpbdp_strp_connect_error' );
+				if ( messageEle ) {
+					messageEle.innerHTML = data.message;
+					messageEle.classList.remove( 'wpbdp-hidden' );
+				}
+				renderStripeConnectSettingsButton();
+			}
+		);
+	}
+
+	function strpSettingsAjaxRequest( action, success, testMode, fail ) {
+		var data = {
+			action: action,
+			testMode: testMode,
+			nonce: wpbdp_global.nonce
+		};
+		postAjax( data, success, fail );
+	}
+
+	function postAjax( data, success, fail ) {
+		var xmlHttp, params;
```

Comment:
`const` would be safe here too.

We could do something like this:
```js
const xmlHttp = new XMLHttpRequest();
const params = typeof data === 'string' ? data : Object.keys( data ).map(
			function( k ) {
				return encodeURIComponent( k ) + '=' + encodeURIComponent( data[ k ]);
			}
		).join( '&' );
```

Although, perhaps we should use jQuery.post() to make the call for consistency and readability.

Reply:
Thanks Sorin!

This is fairly old code, copied from Formidable Stripe.

The main reason it's does with XHR is to avoid jQuery. At least in Formidable, our long term goal is to remove our jQuery dependency.

If we rewrite it, I would opt to use `fetch` instead. I just don't know if it's worth rewriting since it already works.

Reply:
I updated these to const for now.

Reply:
I updated this again to use `fetch` now. It was pretty simple with AI https://github.com/Strategy11/business-directory-plugin/pull/402/commits/7eb7300aa426b69c2474fb7118cd055687a20c86

Reply:
And another commit to simplify it more https://github.com/Strategy11/business-directory-plugin/pull/402/commits/289b9dbb203e9c466a91895d4acd6c303587e5de